pkg/supportbundle/supportbundle.go

```diff
@@ -403,3 +408,37 @@ func CreateSupportBundleAnalysis(appID string, archivePath string, bundle *types
 
 	return nil
 }
+
+func GetSpecSecretsMatchingLabel(client kubernetes.Interface, labelSelector string, namespace string, key string) (*corev1.Secret, error) {
+
+	secrets, err := client.CoreV1().Secrets(namespace).List(context.TODO(), metav1.ListOptions{LabelSelector: labelSelector})
+	if err != nil {
+		return nil, err
+	}
+
+	for _, secret := range secrets.Items {
+		_, ok := secret.Data[key]
+		if !ok {
+			continue
+		}
+		return &secret, nil
+	}
+
+	return nil, kuberneteserrors.NewNotFound(corev1.Resource("secret"), "support bundle spec")
+}
+
+func AddLabelExistingSpecSecret(client kubernetes.Interface, namespace string, secretName string, labels map[string]string) (*corev1.Secret, error) {
```

Comment:
this seems like a generic function that would work with any secret. what do you think about moving it to the `k8sutil` package and renaming it to something like `AddLabelsToSecret`?

Reply:
I have moved that part to k8sutil as AddLabelsToSecret